storage/src/objects/peer.rs

```diff
@@ -0,0 +1,36 @@
+// Copyright (C) 2019-2021 Aleo Systems Inc.
+// This file is part of the snarkOS library.
+
+// The snarkOS library is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// The snarkOS library is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with the snarkOS library. If not, see <https://www.gnu.org/licenses/>.
+
+use std::net::SocketAddr;
+
+use chrono::{DateTime, Utc};
+use serde::{Deserialize, Serialize};
+
+#[derive(Serialize, Deserialize, Clone, Debug)]
+pub struct Peer {
+    pub address: SocketAddr,
+    pub block_height: u32,
+    pub first_seen: Option<DateTime<Utc>>,
+    pub last_seen: Option<DateTime<Utc>>,
+    pub last_connected: Option<DateTime<Utc>>,
+    pub blocks_synced_to: u32,
+    pub blocks_synced_from: u32,
+    pub blocks_received_from: u32,
+    pub blocks_sent_to: u32,
```

Comment:
are these new items designed to give us the ability to check if the peer is not sending too many requests for blocks or blocks to the node and be grounds for a potential ban? we might also want to add something like a `messages_received` to guard against e.g. a flood of `Ping`s or `GetPeers` messages